docs/src/modules/concepts/pages/architecture-model.adoc

```diff
@@ -1,62 +1,70 @@
 
-Domain objects are independent of the other layers. As these are pure Java objects they are the place for implementing business logic. Examples of business logic include checking credit limits for a loan or return policy enforcement. You should write unit tests for this code that tests the business logic. 
+Endpoints use xref:java:component-and-service-calls.adoc#_component_client[`ComponentClient`] to call Akka components in the application layer. This maintains separation of concerns and ensures runtime boundaries are respected.
 
-Much of the inner loop of the developer experience will be spent here. Using Java's `record` type declaration simplifies the amount of ceremony involved in creating domain objects that are understood and used by Akka. This keeps your domain model clean and free of any dependencies - including Akka. 
+The API layer may also expose public event models over Kafka or other channels. External systems should interact with your service only through this layer.
 
-=== Application
+Access control and request validation also belong here. For HTTP-specific guidance, see xref:java:http-endpoints.adoc[Designing HTTP Endpoints].
 
-The application layer is where Akka becomes a first class participant as the glue between your domain code and the runtime. Most classes in this layer will extend an Akka component class (e.g., within the `akka.javasdk` package), making them event-based and actor-based behind the scenes, while abstracting those details from you.
+== Mapping layers to project structure
 
-These components include xref:java:agents.adoc[Agents], xref:java:event-sourced-entities.adoc[Event Sourced Entities], xref:java:key-value-entities.adoc[Key Value Entities],  xref:java:views.adoc[Views], xref:java:workflows.adoc[Workflows], xref:java:timed-actions.adoc[Timers] and xref:java:consuming-producing.adoc[Consumers]. Each component type provides specific functionalities to handle state, events, and interactions efficiently.
+Each layer described above corresponds to a distinct package in the source tree. This structure helps reinforce separation of concerns and makes it easy to locate different types of logic in a project.
 
-=== API
+A typical Akka service might have a layout like the following:
 
-The outermost layer is the API layer which is how the outside world interacts with your application or service. In this layer, you define endpoints that expose your application. Each HTTP endpoint is marked with an `@HttpEndpoint` annotation, which allows the runtime to build the appropriate endpoint URLs for uniquely identifying the components. Each public method on the endpoint that is annotated with method `@Get`, `@Post`, `@Put`, `@Patch` or `@Delete` serves those respective HTTP methods.
+[source,txt]
+----
+src/
+ └── main/
+     └── java/com/example/
+         ├── domain/        # Business logic
+         ├── application/   # Akka components
+         └── api/           # External endpoints
+----
```

Comment:
The structure for a real project is more like:
```
 [org.application-name].[module-name].[api|application|domain]`
 ```
Examples:

* `com.northwind.trading.api`
* `com.northwind.trading.application.StockEntity`
* `com.acme.petclinic.domain.VisitRecord`

Reply:
See https://github.com/akka/akka-sdk/pull/692/files#diff-6b7757175e35345f05f6625d99c8de768523c3866ad003e4d2d6dcd63b7913a7R55